### PR TITLE
[ic-side-navigation]: add unnamed slot functionality and fixed app-title slot bug

### PR DESCRIPTION
**Pull request by j212122**
## Summary of the changes

- Added unnamed slot rendering
- Updated ic-navigation-item/ ic-navigation-group styling
- Fixed app-title slot bug
- Added additional React stories for testing
- Updated ic-nav label styling
- Added visual regression tests
- Added unit and e2e tests
- Updated side-nav animations

## Related issue
#1071 
#845 

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [ ] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [ ] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [ ] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [ ] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [ ] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [ ] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [ ] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue.

### CODE REVIEW

File: packages/web-components/src/components/ic-top-navigation/test/basic/ic-top-navigation.mobile.e2e.ts [hunk 1]
```diff
@@ -77,6 +77,7 @@ describe("ic-top-navigation on mobile", () => {
       <ic-navigation-item
         label="Navigation"
         slot="navigation"
+        href="#"
```
Comment (GCHQ-Developer-465) [inline packages/web-components/src/components/ic-top-navigation/test/basic/ic-top-navigation.mobile.e2e.ts:80]: Is this line supposed to be here? If so, for my understanding what is the purpose of having a href="#"?

File: packages/web-components/src/components/ic-side-navigation/test/basic/ic-side-navigation.e2e.ts [hunk 2]
```diff
@@ -770,5 +772,73 @@ describe("ic-side-navigation", () => {
...
+          .querySelector("ic-side-navigation ic-navigation-item")
+          .querySelector("ic-typography");
+
+        return window.getComputedStyle(label).visibility;
+      });
+
+      expect(visibleLabel).toBe("visible");
+    });
+
+    it("renders visible labels when slotted menu items and expanded", async () => {
```
Comment (GCHQ-Developer-465) [inline packages/web-components/src/components/ic-side-navigation/test/basic/ic-side-navigation.e2e.ts:809]:
```suggestion
    it("renders visible labels when slotted menu items and side nav is expanded", async () => {
```
For consistency thought to add this :)

**Review by GCHQ-Developer-465 — COMMENTED** [review]
Overall, very impressive! Only some small comments :)

**Comment by ad9242** [comment]
In the React  "react router" and "react router with unnamed slot" stories, there's a bit of an alignment issue with the icon\title:

image

Is it just an issue with our examples or more of a component issue?

File: packages/web-components/src/components/ic-navigation-group/ic-navigation-group.tsx [hunk 3]
```diff
@@ -137,23 +137,45 @@ export class NavigationGroup {
...
+        );
+      } else if (this.expanded) {
+        setTimeout(() => {
+          this.expandedNavItemsHeight = this.getNavigationChildItemsHeight();
+
+          this.setGroupedLinksElementHeight(
+            linkWrapper,
+            this.expandedNavItemsHeight
+          );
+        }, 125);
```
Comment (ad9242) [inline packages/web-components/src/components/ic-navigation-group/ic-navigation-group.tsx:160]: Could DYNAMIC_GROUPED_LINKS_HEIGHT_MS be used instead (100ms vs 125ms)?? If not can we add new variable & also use on line 176?

File: packages/web-components/src/components/ic-side-navigation/ic-side-navigation.css [hunk 4]
```diff
@@ -295,8 +323,9 @@ slot[name="app-icon"]::slotted(a) {
 ::slotted(ic-navigation-item),
 ::slotted(ic-navigation-group) {
   --navigation-item-height: 3.5rem;
```
Comment (ad9242) [inline packages/web-components/src/components/ic-side-navigation/ic-side-navigation.css:325]:
```suggestion
```
remove this? as definition duplicated on line 328

File: packages/web-components/src/components/ic-navigation-item/ic-navigation-item.css [hunk 5]
```diff
@@ -367,16 +379,21 @@ svg {
...
+  width: var(--navigation-item-width, auto);
   justify-content: var(--navigation-item-justify-content);
   display: flex;
   gap: var(--ic-space-xs);
   box-sizing: border-box;
   color: var(--navigation-item-child-color) !important;
   text-decoration: none !important;
   white-space: nowrap;
-  padding: 0 var(--ic-space-md);
+  padding-top: var(--ic-space-md) !important;
```
Comment (ad9242) [inline packages/web-components/src/components/ic-navigation-item/ic-navigation-item.css:392]: replace with single padding: to fix lint error

**Review by ad9242 — CHANGES_REQUESTED** [review]

**Comment by ad9242** [comment]
also ran a `npm run lint:fix` to resolve a couple of other errors

**Review by ad9242 — APPROVED** [review]

**Review by GCHQ-Developer-741 — DISMISSED** [review]

**Review by ad9242 — DISMISSED** [review]

**Review by GCHQ-Developer-465 — DISMISSED** [review]

**Review by ad9242 — APPROVED** [review]

**Review by GCHQ-Developer-741 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Pull request merged**